nilearn/input_data/nifti_masker.py

```diff
@@ -200,9 +201,62 @@ def __init__(self, mask_img=None, sessions=None, smoothing_fwhm=None,
         self.memory = memory
         self.memory_level = memory_level
         self.verbose = verbose
+        self.reports = reports
+        self._report_description = ('This report shows the input Nifti '
+                                    'image overlaid with the outlines of the '
+                                    'mask. We recommend to inspect the report '
+                                    'for the overlap between the mask and its '
+                                    'input image. ')
+        self._overlay_text = ('\n To see the input Nifti image before '
+                              'resampling, hover over the displayed image.')
 
         self._shelving = False
 
+    def _reporting(self):
+        """
+        Returns
+        -------
+        displays : list
+            A list of all displays to be rendered.
+        """
+        from .. import plotting
+
+        img, mask = self.input_
+        try:
+            dim = image.load_img(img).shape
+            if len(dim) == 4:
+                # compute middle image from 4D series for plotting
+                img = image.index_img(img, dim[-1] // 2)
+        except TypeError:  # images were not provided to fit
+            img = mask
+
+        # create display of retained input mask, image
+        # for visual comparison
+        init_display = plotting.plot_img(img, black_bg=False)
+        init_display.add_contours(mask, levels=[.5], colors='r')
+
+        if self.transform_ is None:
+            return [init_display]
+
+        else:  # if resampling was performed
+            self._report_description = (self._report_description +
+                                        self._overlay_text)
+
+            # create display of resampled NiftiImage and mask
+            # assuming that resampl_img has same dim as img
+            resampl_img, resampl_mask = self.transform_
+            try:
+                if len(dim) == 4:
+                    # compute middle image from 4D series for plotting
+                    resampl_img = image.index_img(resampl_img, dim[-1] // 2)
+            except NameError:  # images were not provided to fit
```

Comment:
check if `resampl_img is None` rather than catching a `NameError`

Reply:
What's the benefit?